src/protocols/secure_channel/StatusReport.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <protocols/secure_channel/Constants.h>
+#include <protocols/secure_channel/StatusReport.h>
+
+#include <support/BufferReader.h>
+#include <support/CodeUtils.h>
+#include <support/ReturnMacros.h>
+
+#include <type_traits>
+
+using namespace chip::Encoding;
+using GeneralStatusCode = chip::Protocols::SecureChannel::GeneralStatusCode;
+
+namespace chip {
+namespace Protocols {
+namespace SecureChannel {
+
+StatusReport::StatusReport() : mGeneralCode(GeneralStatusCode::kSuccess), mProtocolId(0), mProtocolCode(0), mProtocolData(nullptr)
+{}
+
+StatusReport::StatusReport(GeneralStatusCode generalCode, uint32_t protocolId, uint16_t protocolCode) :
```

Comment:
Please file a followup to use Protocols::Id for the protocol id here and in the StatusReport getter?

Reply:
filed #5573 5573